sktalk/corpus/conversation.py

```diff
@@ -68,3 +69,169 @@ def asdict(self):
             dict: dictionary containing Conversation metadata and Utterances
         """
         return self._metadata | {"Utterances": [u.asdict() for u in self._utterances]}
+
+    def _subconversation_by_index(self,
+                                  index: int,
+                                  before: int = 0,
+                                  after: Optional[int] = None) -> "Conversation":
+        """Select utterances to provide context as a sub-conversation
+
+        Args:
+            index (int): The index of the utterance for which to provide context
+            before (int, optional): The number of utterances prior to indicated utterance. Defaults to 0.
+            after (int, optional): The number of utterances after the indicated utterance. Defaults to None,
+                which then assumes the same value as `before`.
+
+        Raises:
+            IndexError: Index provided must be within range of utterances
+
+        Returns:
+            Conversation: Conversation object without metadata, containing a reduced set of utterances
+        """
+        if index < 0 or index >= len(self._utterances):
+            raise IndexError("Index out of range")
```

Comment:
Maybe I missed the context, but I still wonder if this is necessary. If `index` will be used for retrieving `self._utterances` and it's out of range, the `IndexError` will be thrown anyway, right?

Reply:
Good point! However, because of the way that `before`/`left_bound` and `after`/`right_bound` are calculated, this does not happen. Hence the explicit check..

Reply:
I see, maybe you can do something like this

```suggestion
        assert self._utterances[index]
```

This will still raise an `IndexError` if it's out of range. I make it an assertion just to make this line of code meaningful. You can remove `assert` and do the same trick.

Reply:
Thanks for the suggestion. Indeed, it is off to regurgitate an existing error like this. However, I do like to be explicit here; this is not an error that should happen. I've adjusted the error message now to reflect the context.